apps/event-worker/src/job-execution-dispatch/github.ts

```diff
@@ -77,35 +77,47 @@ export const dispatchGithubJobExecution = async (je: JobExecution) => {
     },
   });
 
-  const { runId, status } = await pRetry(
-    async () => {
-      const runs = await octokit.actions.listWorkflowRuns({
-        owner: config.login,
-        repo: config.repo,
-        workflow_id: config.workflowId,
-      });
-
-      const run = runs.data.workflow_runs.find((run) =>
-        run.name?.includes(je.id.slice(0, 8)),
-      );
-
-      return { runId: run?.id, status: run?.status };
-    },
-    { retries: 15, minTimeout: 1000 },
-  );
-
-  if (runId == null) {
+  let runId: number | null = null;
+  let status: string | null = null;
+
+  try {
```

Comment:
I might be doing something wrong here but if the run is null I need to throw an error so that it triggers the retry. But then if I don't add the outer try catch, typescript thinks that the runID is alwaysdefined, which isn't true because the pRetry could fail 15 times. So I needed to wrap it so that if pRetry fails 15 times, it throws the error and updates the status of the job execution